src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2489,10 +2490,11 @@ class AccessStatementSegment(BaseSegment):
             ),
             "TO",
             OneOf("GROUP", "USER", "ROLE", "SHARE", optional=True),
-            OneOf(
+            Delimited(
                 Ref("ObjectReferenceSegment"),
                 Ref("FunctionSegment"),
                 "PUBLIC",
+                delimiter=Ref("CommaSegment"),
```

Comment:
I'm not sure it's entirely obvious that `Delimited` is the same as `OneOf` (even though it is as you note in #1366). should we wrap this in an actual `OneOf` to make this more obvious (also less to fix if we ever do change `OneOf` to inherit from `Sequence` as you've suggested).

```suggestion
            Delimited(
                OneOf(
                    Ref("ObjectReferenceSegment"),
                    Ref("FunctionSegment"),
                    "PUBLIC",
                ),
                delimiter=Ref("CommaSegment"),
```

There are that many instances of `Delimited` with more than one child, but seems a mixed bag of whether to use a nested OneOf or not (lines 722, 1626, 2088, 2226 in this file do use `OneOf` for example, lines 1397, 1472, 1738 do not).

Personally I think it's clearer if it is in there, even if it's not strictly necessary. WDYT?

Reply:
No strong opinion, so I'll change to that.